### ISSUE TITLE
slow GUI when opening large project - sqlite mapper issue?

### ISSUE THREAD

**Issue by azazellochg** [I2PC/scipion#1423]
When I open a relatively large project with protocols that "have ~300k particles each" it takes 2-3 minutes just to open and then it constantly freezes. I think the problem is heavy sqlite files: I have run several "subset" protocols and for each one there are ~400Mb sqlite files in protocol folder.
If I move/remove a few of these protocol folders the GUI is very quick again.

**Comment by pconesa** [comment]
umm...thanks again @azazellochg, I guess the 400MB db is for the subset not the run.db, am I right?

**Comment by pconesa** [comment]
one more thing......do you think this is due to a recent change? was that project loading fine some time ago? I did some change the way project is loaded....might have caused this...

**Comment by azazellochg** [comment]
@pconesa yes, I meant particle.sqlite files etc. That's a new project so can't tell if this was introduced before. If you could point me to a specific commit that I could revert and check, I can do this :)

**Comment by azazellochg** [comment]
another reason could be that when I open it, the last modified protocol is selected. It could be that calculating/getting protocol summary takes a lot of resources and slows down the GUI?

**Comment by delarosatrevin** [comment]
@azazellochg Any news on this?

**Comment by azazellochg** [comment]
@delarosatrevin I dont have such bug projects anymore, so cannot check this now..

**Comment by delarosatrevin** [comment]
Ok, I was just checking the status of some issues.

**Comment by pconesa** [comment]
I've seen in some cases that loading  the project triggers the "Load" of a set....not for iteration, but for loading it's porperties. For large sets...this is an issue, since for the size we are using end up with a "select count(*) from Objects". This can take even seconds for huge datasets. In one branch I've changed this to use the _size attribute (that it is actually being updated on every Set.append(). Still need to do some test, but this might help. I'm taking this issue to myself.
